testing/e2e/clients/application.go

```diff
@@ -0,0 +1,58 @@
+package clients
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/Azure/aks-app-routing-operator/pkg/util"
+	"github.com/Azure/aks-app-routing-operator/testing/e2e/logger"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
+	msgraphsdk "github.com/microsoftgraph/msgraph-sdk-go"
+	graphapplications "github.com/microsoftgraph/msgraph-sdk-go/applications"
+	graphmodels "github.com/microsoftgraph/msgraph-sdk-go/models"
+)
+
+// GetServicePrincipalOptions populates a new ServicePrincipalOptions struct with fresh credentials and application/client/servicePrincipal object ids
+func GetServicePrincipalOptions(ctx context.Context, applicationObjectID string, credName string) (*ServicePrincipal, error) {
+	lgr := logger.FromContext(ctx)
+	lgr.Info(fmt.Sprintf("getting application with object id %s", applicationObjectID))
+
+	cred, err := getAzCred()
+	scopes := []string{"https://graph.microsoft.com/.default"}
+	graphClient, err := msgraphsdk.NewGraphServiceClientWithCredentials(cred, scopes)
+	if err != nil {
+		return nil, fmt.Errorf("creating graph client: %w", err)
+	}
+
+	getAppResponse, err := graphClient.Applications().ByApplicationId(applicationObjectID).Get(ctx, nil)
+	if err != nil {
+		return nil, fmt.Errorf("getting application with object id %s: %w", applicationObjectID, err)
+	}
+
+	// add new password credential
+	addPasswordReq := graphapplications.NewItemAddPasswordPostRequestBody()
+	newCreds := graphmodels.NewPasswordCredential()
+	newCreds.SetDisplayName(util.StringPtr(credName))
+	newCreds.SetEndDateTime(to.Ptr(time.Now().Add(24 * time.Hour)))
+	addPasswordReq.SetPasswordCredential(newCreds)
+	addPasswordCredResp, err := graphClient.Applications().ByApplicationId(applicationObjectID).AddPassword().Post(ctx, addPasswordReq, nil)
```

Comment:
what permissions does the application have? We should limit it as much as possible. Maybe document the permissions it should have in doc page?

Reply:
the application itself doesn't set perms on creation, so that's down to the service principal later